src/app.jsx

```diff
@@ -494,6 +494,35 @@ const NavigatorCardBody = ({
                 });
             } else if (e.key === "Enter" && selected.length === 1) {
                 onDoubleClickNavigate(selected[0]);
+            } else if (e.key === "Delete") {
+                if (selected.length !== 0) {
+                    const currentDir = path.join("/") + "/";
+                    if (selected.length === 1) {
+                        deleteItem(
+                            Dialogs,
+                            {
+                                selected,
+                                itemPath: currentDir + selected[0].name,
+                                setHistory,
+                                setHistoryIndex,
+                                path: currentDir,
+                                setSelected
+                            }
+                        );
+                    } else {
+                        deleteItem(
+                            Dialogs,
+                            {
+                                selected,
+                                itemPath: currentDir + selected.map(s => currentDir + s.name),
```

Comment:
This doesn't appear to make sense:
```
currentDir + selected.map(s => currentDir + s.name)
'/home/admin/man7//home/admin/man7/EVP_KEYMGMT-Poly1305.7ossl.gz,/home/admin/man7/EVP_KEYMGMT-RSA.7ossl.gz'
```

The code in sidebar.jsx passes `selected` if multiple files are selected.

Reply:
Take a look at the existing `deleteItem` invocation in `src/sidebar.jsx` for [one item](https://github.com/cockpit-project/cockpit-navigator/blob/87e2fc5ca94e0234689e1a485044873afeedfb13/src/sidebar.jsx#L342) and for [multiple](https://github.com/cockpit-project/cockpit-navigator/blob/87e2fc5ca94e0234689e1a485044873afeedfb13/src/sidebar.jsx#L368).

So basically both can be combined and the only difference is that for `selected.length === 1` we need to provide the full path and otherwise itemPath can be left out.

Reply:
I made a last commit. Could I just leave it and work on another issue?

Reply:
There is still an unresolved issue in the PR about combing the if/else into one deleteItem

Reply:
> Actually that's what I was doing exactly in the commit before the last one .but KKoukiou told me that what if multiple files are selected that's why I changed it.so I think we can Ignore the last commit as Mahmoud said above or I still don't know how things work

that's why I replied with this. Anyways I'm going to make another commit